src/main/java/uk/ac/cam/cl/dtg/isaac/api/Constants.java

```diff
@@ -73,7 +73,9 @@ public enum GameboardItemState {
     }
 
     public enum CompletionState {
-        ALL_CORRECT, IN_PROGRESS, NOT_ATTEMPTED
+        ALL_CORRECT, IN_PROGRESS, NOT_ATTEMPTED;
+
+        public static final Set<CompletionState> ALL_STATES = Set.of(CompletionState.values());
```

Comment:
I cannot decide if this is 👌 or 🤢. Clearly the compiler can tell the difference between the enum values and this static attribute, but in the code it's not at all obvious. Should I make it private (I don't want to create it again and again) and use a getter?

Reply:
Tricky. To reduce the ambiguity a little while reading it could maybe be called `ALL_STATES_SET` (although I don't generally like having the name dictate the type)? Like you said compiler/IDE will help make sure it doesn't lead to bugs.
I've also heard that EnumSet is even cooler than Set!

Reply:
A static getter method would probably be the way to go if you want to eliminate ambiguity.